src/IECoreMaya/FromMayaNumericPlugConverter.cpp

```diff
@@ -59,9 +64,40 @@ FromMayaNumericPlugConverter<F,T>::FromMayaNumericPlugConverter( const MPlug &pl
 template<typename F, typename T>
 IECore::ObjectPtr FromMayaNumericPlugConverter<F,T>::doConversion( IECore::ConstCompoundObjectPtr operands ) const
 {
-	F v;
-	plug().getValue( v );
-	return new T( (typename T::ValueType)v );
+	if (plug().isArray())
+	{
+		int numElements = plug().numElements();
+		MIntArray indiceList;
+		typedef TypedData<std::vector<typename T::ValueType> > VecDataType;
+		typename VecDataType::Ptr data = new VecDataType();
+		IntVectorData::Ptr indices = new IntVectorData();
+
+		auto &dataWritable = data->writable();
+		auto &indicesWritable = indices->writable();
+		dataWritable.reserve( numElements );
+		indicesWritable.reserve( numElements );
+
+		for (int i =0; i < numElements; ++i)
+		{
+			MPlug childPlug = plug()[i];
+			F v;
```

Comment:
I think you can just use `typename T::ValueType` here rather than `F` to avoid the cast below, and potential issues with type mismatches.